engine/runner/src/main/java/org/enso/runner/Utils.java

```diff
@@ -0,0 +1,69 @@
+package org.enso.runner;
+
+import java.io.File;
+import java.io.IOException;
+import scala.Tuple3;
+
+final class Utils {
+  private Utils() {}
+
+  /**
+   * Verifies path and project path.
+   *
+   * @param path file or project to execute
+   * @param projectPath project path or {@code null} if it hasn't been specified
+   * @return tuple with boolean, File to execute and path for project to use or {@code null} if
+   *     execution shall finish
+   */
+  static scala.Tuple3<Boolean, File, String> findFileAndProject(String path, String projectPath)
+      throws IOException {
+    var file = new File(path);
+    if (!file.exists()) {
+      System.err.println("File " + file + " does not exist.");
+      return null;
+    }
+    var projectMode = file.isDirectory();
+    var canonicalFile = file.getCanonicalFile();
+    String projectRoot;
+    if (projectMode) {
+      if (projectPath != null) {
+        var canonicalProjectFile = new File(projectPath).getCanonicalFile();
+        if (!canonicalProjectFile.equals(canonicalFile)) {
+          var msg =
+              "It is not possible to run a project ("
+                  + canonicalFile
+                  + ") in context of another "
+                  + "project ("
+                  + canonicalProjectFile
+                  + "), please do not use the `--in-project` option for "
+                  + "running projects.";
```

Comment:
I don't think I understand this message.

`please do not use the `--in-project` option for running projects`? What is this option supposed to be used for, if not exactly that?

Reply:
The message [has been there for ages](https://github.com/enso-org/enso/blob/04b0d266e8cb06fba9ded2ca8a938ee38a7e39ff/engine/runner/src/main/scala/org/enso/runner/Main.scala#L617).

It is activated when the two options `--run project1 --in-project project2` are combined and point to different project.